packages/eslint-plugin-office-addins/test/rules/test-for-null-using-isNullObject.test.ts

```diff
@@ -0,0 +1,126 @@
+import { ESLintUtils } from '@typescript-eslint/experimental-utils'
+import rule from '../../src/rules/test-for-null-using-isNullObject';
+
+const ruleTester = new ESLintUtils.RuleTester({
+  parser: '@typescript-eslint/parser',
+});
+
+const errors = [{ messageId: "useIsNullObject", data: { name: "dataSheet" } }];
+
+ruleTester.run('test-for-null-using-isNullObject', rule, {
+  valid: [ 
+    {
+      code: `await Excel.run(async (context) =>  {
+        var dataSheet = context.workbook.worksheets.getItemOrNullObject("Data");
+        return context.sync().then(function () {
+            if (dataSheet.isNullObject) {
+              dataSheet = context.workbook.worksheets.add("Data");
+            }
+            dataSheet.position = 1;
+         });
+      });`,
+    },
+    {
+      code: `var dataSheet;
+      dataSheet = context.workbook.worksheets.getItemOrNullObject("Data");
+      return context.sync().then(function () {
+        if (dataSheet.isNullObject) {
+          dataSheet = context.workbook.worksheets.add("Data");
+        }
+        dataSheet.position = 1;
+      });`,
+    },
+    {
+      code: `var dataSheet;
+      dataSheet = context.workbook.worksheets.getItemOrNullObject("Data");
+      return context.sync().then(function () {
```

Comment:
Are you checking to context.sync in your code? I don't think it is necessary to your rule and it would be nice to have some tests without it.

I say that because the context.sync might be hidden in some function and I would just like to see some tests without it.

Reply:
The context sync is the technically correct way to use make the call (though there could be other forms I suppose), but will add a mix.